### PR TITLE
Default DataSourceFactory is now mandatory

### PR DESCRIPTION
**Pull request by tutfoo**
- provided a default method for getSecondaryDataSourceFactories so that user does not need to implement
- provided an overridable method to change the reference name for the default primary data source factory
- removed dependency on JDK 1.7 in favor of 1.8

### CODE REVIEW

File: droptools-example/src/main/java/com/bendb/example/ExampleApp.java [hunk 1]
```diff
@@ -33,15 +33,34 @@ public FlywayFactory getFlywayFactory(ExampleConfig configuration) {
...
-            public SortedMap<String,DataSourceFactory> getDataSourceFactories(ExampleConfig configuration) {
+            public DataSourceFactory getDataSourceFactory(ExampleConfig configuration) {
+                return configuration.dataSourceFactoryMaster();
+            }
+
+            /**
+             * Optional override to define secondary DataSourceFactories.
+             */
+            @Override
+            public SortedMap<String,DataSourceFactory> getSecondaryDataSourceFactories(ExampleConfig configuration) {
```
Comment (saem) [inline droptools-example/src/main/java/com/bendb/example/ExampleApp.java:48]: long line

File: dropwizard-jooq/src/main/java/com/bendb/dropwizard/jooq/JooqBundle.java [hunk 2]
```diff
@@ -24,19 +24,21 @@ public void initialize(Bootstrap<?> bootstrap) {
...
-            jooqFactoryConfigurationMap.put(name, cfg);
+        // configure primary data source factory
+        configureDataSourceFactory(
+                configuration,
+                environment,
+                primaryDataSourceName(),
+                getDataSourceFactory(configuration));
+
+        // configure secondary data source factories
+        for (final Map.Entry<String,DataSourceFactory> dataSourceFactoryEntry : getSecondaryDataSourceFactories(configuration).entrySet()) {
```
Comment (saem) [inline dropwizard-jooq/src/main/java/com/bendb/dropwizard/jooq/JooqBundle.java]: Long line, also might want to use the streaming API

File: dropwizard-jooq/src/test/java/com/bendb/dropwizard/jooq/JooqBundleTest.java [hunk 3]
```diff
@@ -59,16 +59,20 @@ public JooqFactory getJooqFactory(DropwizardConfig configuration) {
 
         @Override
         public DataSourceFactory getDataSourceFactory(DropwizardConfig configuration) {
-            return null;
+            return dataSourceFactoryMaster;
         }
 
         @Override
-        public SortedMap<String,DataSourceFactory> getDataSourceFactories(DropwizardConfig configuration) {
+        public SortedMap<String,DataSourceFactory> getSecondaryDataSourceFactories(DropwizardConfig configuration) {
```
Comment (saem) [inline dropwizard-jooq/src/test/java/com/bendb/dropwizard/jooq/JooqBundleTest.java:66]: Long line.

File: dropwizard-jooq/src/main/java/com/bendb/dropwizard/jooq/JooqBundle.java (hunk 2 above)
Comment (tutfoo) [inline dropwizard-jooq/src/main/java/com/bendb/dropwizard/jooq/JooqBundle.java]: good call - fixed

**Pull request merged**